### PR TITLE
Add "whereIsNot" and "whereIsNotAll" scopes to the user model

### PR DESCRIPTION
**Pull request by soltanipoor**

### CODE REVIEW

File: src/Database/Queries/Roles.php [hunk 1]
```diff
@@ -40,6 +56,22 @@ public function constrainWhereIsAll($query, $role)
...
+     * @param  \Illuminate\Database\Eloquent\Builder  $query
+     * @param  string  $role
+     * @return \Illuminate\Database\Eloquent\Builder
+     */
+    public function constrainWhereIsNotAll($query, $role)
+    {
+        $roles = array_slice(func_get_args(), 1);
+
+        return $query->whereHas('roles', function ($query) use ($roles) {
+            $query->whereInNot('name', $roles);
```
Comment (JosephSilber) [inline src/Database/Queries/Roles.php]: `whereNotIn`, no `whereInNot`.

Tests would have caught this. Can't merge this without tests.

Comment (alton01) [inline src/Database/Queries/Roles.php]: hi Joseph
I appreciate you very much for this amazing package 👍 
And thanks for your patience in my mistake.
(sorry for my bad english)

File: src/Database/Queries/Roles.php [hunk 2]
```diff
@@ -40,6 +56,22 @@ public function constrainWhereIsAll($query, $role)
     }
 
     /**
+     * Constrain the given query by all provided roles.
+     *
+     * @param  \Illuminate\Database\Eloquent\Builder  $query
+     * @param  string  $role
+     * @return \Illuminate\Database\Eloquent\Builder
+     */
+    public function constrainWhereIsNotAll($query, $role)
```
Comment (JosephSilber) [inline src/Database/Queries/Roles.php]: I'm really struggling to come up with a scenario where this makes sense.

**Comment by JosephSilber** [comment]
Closed in https://github.com/JosephSilber/bouncer/commit/4f9ee536b7191cec12b664a297124d5d95d31777.

**Pull request closed without merging**